File: pdlearning/src/web-apps/src/libraries/domain-api/src/learner/models/user.model.ts
export interface IUserModel {
  id: string;
  originalUserId: number;
  firstName: string;
  lastName: string;
  email: string;
  departmentId: string;
  isFollowing?: string;
}

export class UserModel implements IUserModel {
  public id: string;
  public originalUserId: number;
  public firstName: string;
  public lastName: string;
  public email: string;
  public departmentId: string;
  public isFollowing?: string;
  constructor(data?: IUserModel) {
    if (!data) {
      return;
    }
    this.id = data.id;
    this.originalUserId = data.originalUserId;
    this.firstName = data.firstName;
    this.lastName = data.lastName;
    this.email = data.email;
    this.departmentId = data.departmentId;
    this.isFollowing = data.isFollowing;
  }

  public get fullName(): string {
    return `${this.firstName} ${this.lastName}`;
  }
}

File: pdlearning/src/microservices/learner-microservice/Api/Application/Models/UserModel.cs
using System;
using Conexus.Opal.AccessControl.Entities;

namespace Microservice.Learner.Application.Models
{
    public class UserModel
    {
        public UserModel(UserEntity entity, bool isFollowing = false)
        {
            Id = entity.Id;
            OriginalUserId = entity.OriginalUserId;
            FirstName = entity.FirstName;
            LastName = entity.LastName;
            Email = entity.Email;
            DepartmentId = entity.DepartmentId;
            IsFollowing = isFollowing;
        }

        public Guid Id { get; set; }

        public int OriginalUserId { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Email { get; set; }

        public int DepartmentId { get; set; }

        public bool IsFollowing { get; set; }
    }
}
